src/libbson/src/bson/bson.c

```diff
@@ -2105,7 +2105,7 @@ bson_new_from_data (const uint8_t *data, size_t length)
 
    BSON_ASSERT (data);
 
-   if ((length < 5) || (length > INT_MAX) || data[length - 1]) {
```

Comment:
But this could be a bug, since `INT_MAX` may differ from `INT32_MAX`/`BSON_MAX_SIZE`

Reply:
Yes, this change is an actual behaviour change. I do believe using INT_MAX is a bug. The largest BSON document size allowed is BSON_MAX_SIZE, but INT_MAX can be much greater than that on a 64-bit architecture.